pylint/checkers/base/basic_checker.py

```diff
@@ -366,20 +366,19 @@ def _check_using_constant_test(
                     call_inferred = inferred.infer_call_result()
                 elif isinstance(inferred, nodes.Lambda):
                     call_inferred = inferred.infer_call_result(node)
```

Comment:
```suggestion
                    call_inferred = list(inferred.infer_call_result(node))
```

Why not just do it in place?

Reply:
Can't it be None ? Or is this just for lambdas ?

Reply:
As far as I know `infer_call_result` should always return a `generator`. If not, that is something we should fix.

I think we can just immediately unpack on L366 and L368.

Reply:
We would get a (false positive I think) message on:
```
    if sched_getaffinity:
        cpu_count = len(sched_getaffinity(0))
```

Reply:
We also need the `list` on line 366

Reply:
I reverted

Reply:
Duh, of course, this work.